src/Nethermind/Nethermind.Consensus.Clique/CliqueBlockProducer.cs

```diff
@@ -351,7 +351,7 @@ public async Task StopAsync()
             }
 
             // Set the correct difficulty
-            header.Difficulty = CalculateDifficulty(snapshot, _address);
+            header.Difficulty = CalculateDifficulty(snapshot, _signer.SigningAddress);
```

Comment:
if you separate roles of signer and sealer (which I do not believe should be the case for Clique) then here you have a first potential issue -> you do not calculate a difficulty of a signer - you calculate difficulty for a particular validator based on their place in the validation order